### PR TITLE
Closes #616 Attribute types Integer and Float are not respected in Property Editor

### LINKED ISSUE #616: Attribute types Integer and Float are not respected in Property Editor.
The attributes are always stored as strings.

Above this core method isValidAttributeValue is not correct at e.g.

https://github.com/webgme/webgme/blob/master/src/common/core/metacore.js#L192

if value is '2fdf' then 2 is returned from parseInt. However shouldn't integer/float attributes be stored as numbers in the first place?

### PR DESCRIPTION
**Pull request by pmeijer**
After this has been merged, we should all agree on what colors to use (if any).

### CODE REVIEW

File: src/client/js/Controls/PropertyGrid/Widgets/IntegerWidget.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    IntegerWidget.prototype.setValue = function (v) {
+
+        WidgetBase.prototype.setValue.call(this, v);
+        this.updateDisplay();
+        return this;
+    };
+
+    IntegerWidget.prototype._onChange = function () {
+        var currentVal = this._input.val();
+        this.setValue(parseInt(currentVal, 10));
```
Comment (lattmann) [inline src/client/js/Controls/PropertyGrid/Widgets/IntegerWidget.js]: I would use `parseFloat` and `Math.round`.

```
parseInt('120e+3', 10)
120
parseFloat('120e+3', 10)
120000
parseFloat('.12e+3', 10)
120
parseInt('.12e+3', 10)
NaN
```

File: src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js [hunk 2]
```diff
@@ -179,548 +172,601 @@ define(['js/logger',
...
+            ptrTo;
 
-                    if (cNode) {
-                        baseNode = _client.getNode(cNode.getBaseId());
-                        ownPointerNames = cNode.getOwnPointerNames();
-                        validNames = cNode.getValidPointerNames();
-                        baseValidNames = baseNode === null ? [] : baseNode.getValidPointerNames();
+        while (len--) {
+            ptrTo = node.getPointer(availablePointers[len]).to;
+            ptrTo = ptrTo === null ? '_nullptr' : ptrTo;
```
Comment (lattmann) [inline src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js]: Is `_nullptr` a constant?

File: src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js [hunk 3]
```diff
@@ -179,548 +172,601 @@ define(['js/logger',
...
+            range = {},
+            schema;
 
-                        if (validNames.indexOf(pointerName) !== -1 || NON_INVALID_PTRS.indexOf(pointerName) !== -1) {
-                            return false;
-                        }
-                    }
+        while (i--) {
+            schema = this._client.getAttributeSchema(selectedObjIDs[i], attrName);
+            if (schema.hasOwnProperty('min')) {
```
Comment (lattmann) [inline src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js]: Are `min` and `max` constants?

File: src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js [hunk 4]
```diff
@@ -179,548 +172,601 @@ define(['js/logger',
...
+                        } else if (key === REGISTRY_KEYS.SVG_ICON || key === REGISTRY_KEYS.PORT_SVG_ICON) {
+                            dst[extKey].widget = PropertyGridWidgets.DIALOG_WIDGET;
+                            dst[extKey].dialog = DecoratorSVGExplorerDialog;
+                        }
+                    }
+                } else if (prefix === CONSTANTS.PROPERTY_GROUP_META + '.') {
+                    if (key === REGISTRY_KEYS.VALID_VISUALIZERS) {
+                        dst[extKey].value = dst[extKey].value === undefined ?
+                            '' : dst[extKey].value;
+                        dst[extKey].regex = '/[^\w\W]/';
```
Comment (lattmann) [inline src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js]: This regex looks suspicious.

File: src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js (hunk 2 above)
Comment (pmeijer) [inline src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js]: Yes, that code was there before but I can change it in this branch.

File: src/client/js/Controls/PropertyGrid/Widgets/IntegerWidget.js (hunk 1 above)
Comment (pmeijer) [inline src/client/js/Controls/PropertyGrid/Widgets/IntegerWidget.js]: Good point

File: src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js (hunk 3 above)
Comment (pmeijer) [inline src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js]: No, currently not.

File: src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js (hunk 4 above)
Comment (pmeijer) [inline src/client/js/Panels/PropertyEditor/PropertyEditorPanelController.js]: It's an old workaround for displaying the available plugins, visualizer etc. I'll make an issue for a multi-select space separated string value.

**Pull request merged**